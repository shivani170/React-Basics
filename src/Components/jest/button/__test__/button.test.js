import React from 'react';
import ReactDom from 'react-dom';
import Button from './../button';
import { render, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';

afterEach(cleanup);

it("render without crashing", () => {
    const div = document.createElement("div");
    ReactDom.render(<Button />, div)
})

it("renders button correctly", () => {
    const { getByTestId} = render(<Button label="Click me"/>)
    expect(getByTestId('button')).toHaveTextContent("Click me")
})

it("renders button save correctly", () => {
    const { getByTestId} = render(<Button label="Save"/>)
    expect(getByTestId('button')).toHaveTextContent("Save")
})

// it("matches snapshot", ()=>{
//     const tree = renderer.create(<Button  label="Please Save"/>).toJSON()
//     expect(tree).toMatchSnapshot()
// })