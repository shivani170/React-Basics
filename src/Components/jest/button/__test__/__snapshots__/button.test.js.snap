// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
Array [
  <div>
    My Counter
  </div>,
  <div
    className="button-style"
    data-testid="button"
  >
    Please Save
  </div>,
]
`;
